eventsources/sources/amqp/start.go

```diff
@@ -155,24 +161,93 @@ func (el *EventListener) StartListening(ctx context.Context, dispatch func([]byt
 	}
 }
 
+// setDefaults sets the default values in case the user hasn't defined them
+// helps also to keep retro-compatibility with current dpeloyments
+func setDefaults(eventSource *v1alpha1.AMQPEventSource) error {
```

Comment:
It looks good to me except a minor: there's no chance/need to return error in this function. Also, please confirm it is backward compatible with real testing - I don't have an AMQP env setup, can not do a test :), thanks!

Reply:
I tested this branch using the example on the website and it works without issues. The messages below are a confirmation
```
{"level":"info","ts":1610568336.564611,"logger":"argo-events.eventsource","caller":"eventsources/eventing.go:350","msg":"succeeded to publish an event","eventSourceName":"amqp","eventName":"example","eventSourceType":"amqp"}
{"level":"info","ts":1610568406.6777697,"logger":"argo-events.eventsource","caller":"amqp/start.go:114","msg":"received the message","eventSourceName":"amqp","eventSourceType":"amqp","eventName":"example","message-id":""}
{"level":"info","ts":1610568406.6780057,"logger":"argo-events.eventsource","caller":"amqp/start.go:143","msg":"dispatching event ...","eventSourceName":"amqp","eventSourceType":"amqp","eventName":"example"}
{"level":"info","ts":1610568406.6828492,"logger":"argo-events.eventsource","caller":"eventsources/eventing.go:350","msg":"succeeded to publish an event","eventSourceName":"amqp","eventName":"example","eventSourceType":"amqp"}
```
If somebody encounters an issue, I'll be happy to fix it 😄